netket/utils/static_range.py

```diff
@@ -225,12 +246,12 @@ def __eq__(self, o):
             )
         elif hasattr(o, "shape"):
             if self.shape == o.shape:
-                return self.__array__() == o
+                return np.all(self.__array__() == o)
         elif hasattr(o, "__array__"):
             if hasattr(o, "shape") and self.shape == o.shape:
-                return self.__array__() == o
+                return np.all(self.__array__() == o.__array__())
```

Comment:
Those were defined like that explicitly. 
Array comparison is defined to be element wise.
That's why I was checking for same shape.
The user can call `np.all` if he wants.

Reply:
OK, sorry, I didn't know. I will change it back.

Reply:
(it's not my decision, it's numpty's default, but since those objects behave like arrays that's what I decided to implement)